app/scripts/modules/core/src/artifact/expectedArtifact.service.ts

```diff
@@ -104,13 +102,13 @@ export class ExpectedArtifactService {
 
   public static getKindConfig(artifact: IArtifact, isDefault: boolean): IArtifactKindConfig {
     if (artifact == null || artifact.customKind || artifact.kind === 'custom') {
-      return Registry.pipeline.getCustomArtifactKind();
```

Comment:
Given I have an artifact that is already defined with a `customKind` field or a `kind` set to `'custom'`, this change results in my artifact being dropped from the expected-artifacts section in the config UI. This happens because `app/scripts/modules/core/src/pipeline/config/triggers/artifacts/artifact.component.ts` calls out to `getKindConfig()` to determine which template to render for a given artifact.

Reply:
Before:

![screen shot 2019-03-04 at 12 43 29 pm](https://user-images.githubusercontent.com/34253460/53751819-25590000-3e7b-11e9-91be-4b98b7bef558.png)

After:

![screen shot 2019-03-04 at 12 42 10 pm](https://user-images.githubusercontent.com/34253460/53751825-2b4ee100-3e7b-11e9-9cc7-34c5532cb8c5.png)

Reply:
Restored in 5ab4885a4. There was more to do afterward. Turns out that the old angular custom editor was calling a special function `Registry.pipeline.registerCustomKind`. That was essential to the custom editor showing up here. Since I'm suggesting in 1.14 that we do away with the custom editor entirely, I've just removed the new Custom react component and reinstated the angular component.